### PR TITLE
Log4j update and remove maven-scr plugin

### PR DESCRIPTION
**Pull request by madushadhanushka**
Remove maven-src plugin
update kernel version
update with log4j2 changes
fix review changes

### CODE REVIEW

File: components/ganalytics/pom.xml [hunk 1]
```diff
@@ -36,10 +36,10 @@
             <groupId>org.wso2.orbit.org.apache.httpcomponents</groupId>
             <artifactId>httpclient</artifactId>
         </dependency>
-        <dependency>
-            <groupId>commons-logging</groupId>
-            <artifactId>commons-logging</artifactId>
-        </dependency>
+      <dependency>
+    <groupId>org.ops4j.pax.logging</groupId>
```
Comment (jsdjayanga) [inline components/ganalytics/pom.xml]: Alignment issue

File: service-stubs/url-mapper/pom.xml [hunk 2]
```diff
@@ -73,9 +73,9 @@
             <groupId>org.apache.ws.commons.schema.wso2</groupId>
             <artifactId>XmlSchema</artifactId>
         </dependency>
-        <dependency>
-            <groupId>commons-logging</groupId>
-            <artifactId>commons-logging</artifactId>
-        </dependency>
+      <dependency>
```
Comment (madushadhanushka) [inline service-stubs/url-mapper/pom.xml]: Alignment issue

File: components/logging/org.wso2.carbon.logging.updater/src/main/java/org/wso2/carbon/logging/updater/LogConfigUpdater.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        try (WatchService watchService = FileSystems.getDefault().newWatchService()) {
+            Path carbonConfigDirectory = Paths.get(carbonConfigDirPath);
+            carbonConfigDirectory.register(watchService, StandardWatchEventKinds.ENTRY_MODIFY);
+            WatchKey key;
+            while ((key = watchService.take()) != null) {
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    WatchEvent<Path> watchEvent = (WatchEvent<Path>) event;
+                    Path fileName = watchEvent.context();
+                    Path changedFile = carbonConfigDirectory.resolve(fileName);
+                    if (changedFile.toFile().equals(log4j2File)) {
```
Comment (jsdjayanga) [inline components/logging/org.wso2.carbon.logging.updater/src/main/java/org/wso2/carbon/logging/updater/LogConfigUpdater.java]: Shall we try adding CanonicalPath here?

File: features/logging-mgt/org.wso2.carbon.logging.updater.feature/pom.xml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!--
- ~ Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ ~ Copyright (c) 2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jsdjayanga) [inline features/logging-mgt/org.wso2.carbon.logging.updater.feature/pom.xml]: Should be 2019

File: features/logging-mgt/org.wso2.carbon.logging.updater.feature/pom.xml [hunk 5]
```diff
@@ -25,60 +25,54 @@
     </parent>
 
     <modelVersion>4.0.0</modelVersion>
-    <artifactId>org.wso2.carbon.logging.mgt.ui.feature</artifactId>
+    <artifactId>org.wso2.carbon.logging.updater.feature</artifactId>
     <packaging>pom</packaging>
-    <name>WSO2 Carbon - Logging Management UI Feature</name>
+    <name>WSO2 Carbon - SSO Hostobjects Feature</name>
```
Comment (jsdjayanga) [inline features/logging-mgt/org.wso2.carbon.logging.updater.feature/pom.xml]: This should be something related to Updater

File: service-stubs/ndatasource/org.wso2.carbon.ndatasource.stub/pom.xml [hunk 6]
```diff
@@ -113,11 +113,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <dependency>
-            <groupId>commons-logging</groupId>
-            <artifactId>commons-logging</artifactId>
-        </dependency>
-        <dependency>
+		<dependency>
```
Comment (jsdjayanga) [inline service-stubs/ndatasource/org.wso2.carbon.ndatasource.stub/pom.xml]: Alignment issue

File: service-stubs/remote-tasks/pom.xml [hunk 7]
```diff
@@ -73,9 +73,9 @@
             <groupId>org.apache.ws.commons.schema.wso2</groupId>
             <artifactId>XmlSchema</artifactId>
         </dependency>
-        <dependency>
-            <groupId>commons-logging</groupId>
-            <artifactId>commons-logging</artifactId>
-        </dependency>
+      <dependency>
```
Comment (jsdjayanga) [inline service-stubs/remote-tasks/pom.xml]: Alignment issue

File: service-stubs/reporting/pom.xml [hunk 8]
```diff
@@ -74,9 +74,9 @@
             <groupId>org.apache.ws.commons.schema.wso2</groupId>
             <artifactId>XmlSchema</artifactId>
         </dependency>
-        <dependency>
-            <groupId>commons-logging</groupId>
-            <artifactId>commons-logging</artifactId>
-        </dependency>
+      <dependency>
```
Comment (jsdjayanga) [inline service-stubs/reporting/pom.xml]: Alignment issue

**Pull request merged**